packages/server-wallet/src/protocols/__test__/ledger-manager.test.ts

```diff
@@ -172,33 +172,81 @@ describe('marking ledger requests as complete', () => {
+      // So here I interpret "takes no action" <> no new signed state in the outbox
+
+      // create a ledger channel. Note distinct channel nonce ensured automatically
+      const ledgerChannel = TestLedgerChannel.create({});
+      await ledgerChannel.insertInto(store, {
+        states: [5, 6],
+      });
+      // insert a ledger proposal identical to the existing outcome
+      await ledgerChannel.insertProposal(
+        store,
+        ledgerChannel.startOutcome, // proposal
+        ledgerChannel.participantA.signingAddress,
+        1
+      );
+
+      // crank the ledger manager
+      const response = new WalletResponse();
+      await ledgerManager.crank(ledgerChannel.channelId, response);
+
+      // assert that the signed state in the outbox has the same turnNum (we are just pessimistically resending)
+      expect(
+        (response.multipleChannelOutput().outbox[0]?.params.data as any).signedStates[0].turnNum
+      ).toEqual(6);
     });
 
-    it('takes no action if proposal exists but counterparties does not', () => {
-      const requestChannel = prefundChannelWithAllocations([alice, 1]);
-      const protocolArgs = {
-        fundingChannel: defaultLedgerChannel,
-        channelsRequestingFunds: [requestChannel.protocolState],
-        myLedgerProposal: {
-          proposal: simpleEthAllocation([alice, 9], [bob, 10], [requestChannel.channelId, 1]),
-          nonce: 0,
-        },
-      };
-      expect(protocol(processLedgerQueueProtocolState(protocolArgs))).toBeUndefined();
+    it(`takes no action if proposal exists but counterparty's does not`, async () => {
+      // NOTE: currently, there are some side effects that happen even when there is "no action"
+      // these side effects are the "pessimistic" resending of proposals and states.
+      // So here I interpret "takes no action" <> no new signed state in the outbox
+
+      // create an application channel
+      const appChannel = TestChannel.create({aBal: 1, bBal: 0});
+      await appChannel.insertInto(store, {states: [0, 1]});
+      // create a ledger channel whose current state doesn't fund that channel. Note distinct channel nonce ensured automatically
+      const ledgerChannel = TestLedgerChannel.create({});
+      await ledgerChannel.insertInto(store, {
+        // Will be "fully funded" i.e. with 10 coins
+        states: [4, 5],
+        bals: [
+          [appChannel.participantA.destination, 10],
+          [appChannel.participantB.destination, 10],
+        ],
+      });
+      // create a ledger request for the ledger to fund the channel
+      await ledgerChannel.insertFundingRequest(store, appChannel.channelId);
+      // insert a ledger proposal identical to the existing outcome
+      await ledgerChannel.insertProposal(
+        store,
+        simpleEthAllocation(
+          [appChannel.participantA.destination, 9],
+          [appChannel.participantB.destination, 10],
+          [appChannel.channelId, 1]
+        ), // proposal
```

Comment:
Is this how proposals work? I thought the proposal would be just `[appChannel.channelId, 1]`, otherwise it seems like it would be difficult to have multiple simultaneous proposals.

Reply:
I don't know. But as this is just a refactor, I've kept it identical to the existing test.